src/renderers/abstract.renderer.ts

```diff
@@ -763,5 +775,6 @@ export abstract class Renderer {
     return doc.documentElement.outerHTML
   }
 
+  abstract download(downloadOpts: DownloadOpts): Promise<DownloadRes>
```

Comment:
Why are we using this pattern of one parameter that is a typed object, rather than just using normal typed parameters? Ie:

```
abstract download(downloader: Downloader, articleUrl: string, articleDetail: ArticleDetail)
```

Reply:
I have no strong opinions on this